pkg/asset/machines/aws/machines.go

```diff
@@ -0,0 +1,121 @@
+// Package aws generates Machine objects for aws.
+package aws
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/pointer"
+	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1alpha1"
+	clusterapi "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Machines returns a list of machines for a machinepool.
+func Machines(config *types.InstallConfig, pool *types.MachinePool, role, userDataSecret string) ([]clusterapi.Machine, error) {
+	var out []clusterapi.Machine
```

Comment:
You can drop this declaration here and use `(out []clusterapi.Machine, err error)` in the line above (I also prefer `machines` to `out`).  But both of those are minor nits.